File: problems/55/solution.go
func canJump(nums []int) bool {
    leftmostGoodIndex := len(nums) - 1
    
    for currentPosition := len(nums) - 2; currentPosition >= 0; currentPosition-- {
        if currentPosition + nums[currentPosition] >= leftmostGoodIndex {
            leftmostGoodIndex = currentPosition
        }
    }
    
    return leftmostGoodIndex == 0
}

File: problems/55/solution.cs
public class Solution {
    public bool CanJump(int[] nums) {
        var leftmostGoodIndex = nums.Length - 1;
        
        for (var i = nums.Length - 2; i >= 0; i--) {
            if (i + nums[i] >= leftmostGoodIndex) {
                leftmostGoodIndex = i;
            }
        }
        
        return leftmostGoodIndex == 0;
    }
}